### PR TITLE
Support records in [Add/Show/Drop/Rename]Columns functions

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
Allow those functions to work in records in addition to tables.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs [hunk 1]
```diff
@@ -52,17 +63,31 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             Contracts.AssertValue(errors);
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
-            var fArgsValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
-            Contracts.Assert(returnType.IsTable);
+            returnType = ReturnType;
+            nodeToCoercedTypeMap = null;
 
-            if (!argTypes[0].IsTable)
+            var isRecord = false;
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs:69]: This code seems pretty shared, should Show/Drop columns be refactored to share a base class?

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs]: Good idea, done

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by anderson-joyle** [comment]
> #SETUP: SupportColumnNamesAsIdentifiers

Do we need another txt file testing `ShowColumns` with `disable:SupportColumnNamesAsIdentifiers`?

---
Refers to: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ShowColumns.txt:1 in 71ef6fc. [](commit_id = 71ef6fc8f55fe278c3663e066f75273c10ee52cc, deletion_comment = False)

**Review by anderson-joyle — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+                    // Argument '{0}' is not a valid identifier.
+                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, expectedColumnName);
+                    continue;
+                }
+
+                var columnName = _isShowColumns ?
+                    new DName(expectedColumnName) :
+                    new DName(
+                        DType.TryGetLogicalNameForColumn(argTypes[0], expectedColumnName, out var logicalName)
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs:161]: >DType.TryGetLogicalNameForColumn(argTypes[0], expectedColumnName, out var logicalName) [](http://example.com/codeflow?start=24&length=86)

Display names should work for both `DropColumns` and `ShowColumns`.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs (hunk 2 above)
Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs]: Updated

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, expectedColumnName);
+                    continue;
+                }
+
+                var columnName = new DName(
+                    DType.TryGetLogicalNameForColumn(argTypes[0], expectedColumnName, out var logicalName)
+                    ? logicalName
+                    : expectedColumnName);
+
+                // Verify that the name exists.
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs:163]: >// Verify that the name exists. [](http://example.com/codeflow?start=16&length=31)

Wrong comment?

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs:163]: Good catch, it's misplaced. Updated.

**Review by anderson-joyle — APPROVED** [review]
:shipit:

**Pull request merged**